### PR TITLE
Fix docker image download progress

### PR DESCRIPTION
**Pull request by msanjeevkumar**

### CODE REVIEW

**Review by sekulicd — COMMENTED** [review]
I think this is right way to go, to find progress percentage based on bellow info returned by docker engine:
`'progressDetail': {'current': X, 'total': Y},`

However there is an error thrown when last message is sent from docker engine:
`{'status': 'Pull complete', 'progressDetail': {}, 'id': '{SOME_ID}'}`

Error Stack Trace:
ERROR:    Exception in ASGI application
```
Traceback (most recent call last):
  File "/Users/sekulicd/.pyenv/versions/3.11.2/lib/python3.11/site-packages/uvicorn/protocols/http/h11_impl.py", line 429, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
.
.
.
    async for data in self.body_iterator:
  File "/Users/sekulicd/PycharmProjects/prem-daemon/app/routes.py", line 181, in generator
    layer_id = line["id"]
               ~~~~^^^^^^
```

@msanjeevkumar should you check if status=='Pull complete' or 'progressDetail': {}  and exit?

**Comment by msanjeevkumar** [comment]
@sekulicd please pull latest, i have already fixed the `pull complete` issue in https://github.com/premAI-io/prem-daemon/commit/16a7d265d6311dc20f44d5bbca4fcc644ddb99c5

**Comment by msanjeevkumar** [comment]
@sekulicd  Fixed the pull complete issue as well, please review

File: app/routes.py [hunk 1]
```diff
@@ -149,26 +161,37 @@ def generator():
...
+        Status.WAITING: lambda _: 0,
+        Status.VERIFYING_CHECKSUM: lambda _: 97,
+        Status.DOWNLOAD_COMPLETE: lambda _: 98,
+        Status.EXTRACTING: lambda _: 99,
+        Status.PULL_COMPLETE: lambda _: 100,
+    }
+    layers = {}
 
     client = utils.get_docker_client()
+
```
Comment (sekulicd) [inline app/routes.py:177]: Please handle exception(with try/except) when invoking docker.

File: app/routes.py [hunk 2]
```diff
@@ -149,26 +161,37 @@ def generator():
 
 
 async def generator(service_object, request):
-    total_downloaded = 0
-    partial_downloaded = 0
+    progress_mapping = {
```
Comment (sekulicd) [inline app/routes.py:164]: Please move constant outside of func.

File: app/routes.py (hunk 1 above)
Comment (msanjeevkumar) [reply app/routes.py:177]: this is the original code, this is how docker is getting invoked in code base

File: app/routes.py (hunk 2 above)
Comment (msanjeevkumar) [reply app/routes.py]: Done

**Comment by sekulicd** [comment]
tACK

**Comment by filopedraz** [comment]
@msanjeevkumar thanks for the contribution

**Pull request merged**